### PR TITLE
add multi_company related modules (TA#20777)

### PR DESCRIPTION
**Pull request by foutoucour**
https://numigi.com/web?debug=1#id=20777&action=298&model=project.task&view_type=form&menu_id=200

### CODE REVIEW

**Comment by foutoucour** [comment]
@ddufresne je voudrais ton avis sur ces modules.
Ce sont des modules de cybrosis (https://apps.odoo.com/apps/modules/12.0/hr_multi_company/) pour une tâche pour cosmos. Mais ils ne sont pas nécessairement lié au métier de cosmos.
Du coup je voulais les ajouter ici. Qu'est ce que tu en penses ?

Un nouveau repo me paraissait overkill

File: hr_employee_transfer/models/res_company.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+        for company in obj_company:
+            obj_branch = self.env['transfer.company'].search([('company_id', '=', company.id)])
+            com = {}
+            if not obj_branch:
+                com = {
+                    'name': company.name,
+                    'company_id': company.id,
+                }
+                obj = self.env['transfer.company'].create(com)
```
Comment (foutoucour) [inline hr_employee_transfer/models/res_company.py:28]: Codacy Issue found: [Unused variable 'obj'](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5137145)

File: hr_employee_transfer/models/hr_employee.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+from odoo import models, fields, api
```
Comment (foutoucour) [inline hr_employee_transfer/models/hr_employee.py:2]: Codacy Issue found: [Unused api imported from odoo](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5137145)

File: hr_multi_company/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+###################################################################################
+{
```
Comment (foutoucour) [inline hr_multi_company/__manifest__.py:23]: Codacy Issue found: [Statement seems to have no effect](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5137145)

File: hr_employee_transfer/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+###################################################################################
+{
```
Comment (foutoucour) [inline hr_employee_transfer/__manifest__.py:23]: Codacy Issue found: [Statement seems to have no effect](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5137145)

File: hr_employee_transfer/models/employee_transfer.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+from datetime import date
+from odoo import models, fields, api, _
+from odoo.tools import DEFAULT_SERVER_DATE_FORMAT
+from odoo.exceptions import Warning
```
Comment (foutoucour) [inline hr_employee_transfer/models/employee_transfer.py:5]: Codacy Issue found: [Redefining built-in 'Warning'](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5137145)

File: hr_employee_updation/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+###################################################################################
+{
```
Comment (foutoucour) [inline hr_employee_updation/__manifest__.py:23]: Codacy Issue found: [Statement seems to have no effect](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5137145)

File: hr_employee_transfer/models/employee_transfer.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+                self.wage = obj_contract.wage
+        self.state = 'transfer'
+        self.employee_id = new_emp
+        obj_emp.write({'active': False})
+
+    @api.multi
+    def receive_employee(self):
+        for this in self:
+            if this._context is None:
+                context = {}
```
Comment (foutoucour) [inline hr_employee_transfer/models/employee_transfer.py:80]: Codacy Issue found: [Unused variable 'context'](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5137145)

File: hr_employee_transfer/models/employee_transfer.py [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    def transfer(self):
+        obj_emp = self.env['hr.employee'].browse(self.employee_id.id)
+        emp = {}
+        if not self.branch:
+            raise Warning(_(
+                'You should select the transfer branch/company.'))
+        if self.branch.company_id == self.company_id.id:
+            raise Warning(_(
+                'You cant transfer to same company.'))
+        for this in self:
```
Comment (foutoucour) [inline hr_employee_transfer/models/employee_transfer.py:57]: Codacy Issue found: [Unused variable 'this'](https://app.codacy.com/manual/numigi/odoo-hr-addons/pullRequest?prid=5137145)

**Review by ddufresne — CHANGES_REQUESTED** [review]
@foutoucour stp ne pas polluer nos répertoires de modules avec les modules de cybrosis

On a déjà un répertoire pour les modules cybrosis
https://github.com/Numigi/odoo-cybrosis-addons

File: hr_multi_company/__manifest__.py [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    'name': 'Open HRMS Multi-Company',
+    'version': '12.0.1.0.0',
+    'summary': """Enables Multi-Company""",
+    'description': 'This module enables multi company features',
+    'category': 'Generic Modules/Human Resources',
+    'author': 'Cybrosys Techno Solutions',
+    'company': 'Cybrosys Techno Solutions',
+    'website': "https://www.openhrms.com",
+    'depends': ['base', 'hr_contract', 'hr_payroll', 'hr_expense', 'hr_attendance', 'hr_employee_transfer'],
```
Comment (ddufresne) [inline hr_multi_company/__manifest__.py:32]: @foutoucour ce module est un monolith. Il dépend de toutes les applications hr_*.

**Pull request closed without merging**